tests/data_collector/test_data_collector.py

```diff
@@ -176,21 +174,22 @@ def test_reproducibility(seed):
     # Step through the env again using the stored seed and check it matches
     env = dataset.recover_environment()
 
-    for episode in dataset.sample_episodes(dataset.total_episodes):
+    for episode in dataset.iterate_episodes():
         if seed is None:
             assert isinstance(episode.seed, int)
             assert episode.seed >= 0
         else:
             assert seed == episode.seed
 
         obs, _ = env.reset(seed=episode.seed)
-        env.action_space.seed(policy_seed)
 
         assert np.allclose(obs, episode.observations[0])
 
         for k in range(episode.total_timesteps):
-            obs, rew, _, _, _ = env.step(env.action_space.sample())
+            obs, rew, term, trunc, _ = env.step(episode.actions[k])
             assert np.allclose(obs, episode.observations[k + 1])
             assert rew == episode.rewards[k]
+            assert not term
+            assert trunc == (k == episode.total_timesteps - 1)
```

Comment:
does something like
```python
assert term == episode.terminations[k]
assert trunc == episode.truncations[k]
```

works?

Reply:
Sorry, it does of course. Thanks for spoting this!

Reply:
No worries, thanks indeed, looks good!